Kitodo/src/main/java/org/kitodo/production/model/Subfolder.java

```diff
@@ -101,12 +102,15 @@ public Subfolder(Process process, Folder folder) {
      */
     private static Function<URI, String> createKeyMapperForPattern(final Pattern pattern) {
         return uri -> {
-            Matcher matcher = pattern.matcher(FilenameUtils.getName(uri.getPath()));
-            if (!matcher.matches()) {
-                throw new IllegalStateException(
-                        "At this point may only arrive files where the pattern had already matched.");
+            if(Objects.nonNull(uri)) {
```

Comment:
Missing white space:
```suggestion
            if (Objects.nonNull(uri)) {
```